promgen/static/js/promgen.vue.js

```diff
@@ -323,3 +351,95 @@ app.component('exporter-test', {
         }
     }
 });
+
+app.component('silence-table-modal', {
+    template: '#silence-table-modal-template',
+    delimiters: ['[[', ']]'],
+    mixins: [mixins],
+    data() {
+        return {
+            state: silenceTableStore.state,
+            form: {
+                labels: [],
+                label: '',
+                value: ''
+            },
+            store: dataStore
+        };
+    },
+    computed: {
+        globalMessages() {
+            return globalStore.state.messages;
+        },
+        activeSilences() {
+            return this.$root.activeSilences || [];
+        },
+        filteredSilences() {
+            if (!this.form.labels || this.form.labels.length === 0) {
+                return this.activeSilences;
+            }
+            
+            return this.activeSilences.filter(silence => {
+                return this.form.labels.some(filterLabel => {
+                    return silence.matchers.some(matcher => 
+                        matcher.name === filterLabel.label &&
+                        matcher.value === filterLabel.value
+                    );
+                });
+            }); 
+        }
+    },
+    methods: {
+        hideModal() {
+            const modal = $('#silenceTableModal');
+            if (modal.length) {
+                globalStore.setMessages([]);
+                this.form.labels = [];
+                modal.modal('hide');
+            }
+        },
+        showModal() {
+            const modal = $('#silenceTableModal');
+            if (modal.length) {
+                modal.on('hidden.bs.modal', () => {
+                    silenceTableStore.hideModal();
+                });
+                modal.modal('show');
+            }
+        },
```

Comment:
In this case, $ is indeed jQuery, not something built into Vue. We're using it here because our Modal is still using Bootstrap 3, which relies on jQuery for DOM manipulation and event triggering.